pkg/cmd/pulumi/destroy.go

```diff
@@ -134,20 +149,11 @@ func newDestroyCmd() *cobra.Command {
 				return result.FromError(fmt.Errorf("gathering environment metadata: %w", err))
 			}
 
-			sm, err := getStackSecretsManager(s)
-			if err != nil {
-				return result.FromError(fmt.Errorf("getting secrets manager: %w", err))
-			}
-
-			cfg, err := getStackConfiguration(s, sm)
-			if err != nil {
-				return result.FromError(fmt.Errorf("getting stack configuration: %w", err))
-			}
-
```

Comment:
moved lower to use the snapshot's secretsManager

Reply:
Is it that `getStackSecretsManager(s)` fails in the `pulumi destroy -s <stack>` scenario out of dir?

Reply:
I think we need to decrypt secrets in order to get the config to run the engine.